asabiris/handlers/kafkahandler.py

```diff
@@ -175,72 +182,86 @@ async def send_email(self, json_data):
+
+		except Exception as e:
+			# Log any unexpected exceptions that might occur
+			L.warning("Following unexpected error occurred while handling exception: {}".format(str(e)))
 
 	def generate_error_message(self, specific_error: str, service_type: str):
-		timestamp = datetime.datetime.now(tz=datetime.timezone.utc).strftime("%Y-%m-%d %H:%M:%S")
-
-		if service_type == 'email':
-			error_subject = "Error when generating notification"
-			error_message = (
-				"<p>Hello!</p>"
-				"<p>We encountered an issue while processing your request:<br><b>{}</b></p>"
-				"<p>Please review your input and try again.<p>"
-				"<p>Time: {} UTC</p>"
-				"<p>Best regards,<br>Your Team</p>"
-			).format(specific_error, timestamp)
-			return error_message, error_subject
-
-		elif service_type == 'slack':
-			error_message = (
-				":warning: *Hello!*\n\n"
-				"We encountered an issue while processing your request:\n`{}`\n\n"
-				"Please review your input and try again.\n\n"
-				"*Time:* `{}` UTC\n\n"
-				"Best regards,\nYour Team :robot_face:"
-			).format(specific_error, timestamp)
-			return error_message, None
-
-
-		elif service_type == 'msteams':
-			error_message = (
-				"Warning: Hello!\n\n"
-				"We encountered following issue while processing your request.\n\n`{}`\n\n"
-				"Please review your input and try again.\n\n"
-				"Time: `{}` UTC\n\n"
-				"Best regards,\nYour Team"
-			).format(specific_error, timestamp)
-			return error_message, None
+		try:
+			timestamp = datetime.datetime.now(tz=datetime.timezone.utc).strftime("%Y-%m-%d %H:%M:%S")
+
+			if service_type == 'email':
+				error_subject = "Error Generating Email Notification"
+				error_message = (
+					"<p>Hello!</p>"
+					"<p>We encountered an issue while processing your request:<br><b>{}</b></p>"
+					"<p>Please review your input and try again.<p>"
+					"<p>Time: {} UTC</p>"
+					"<p>Best regards,<br>Your Team</p>"
+				).format(specific_error, timestamp)
+				return error_message, error_subject
+
+			elif service_type == 'slack':
+				error_message = (
+					":warning: *Hello!*\n\n"
+					"We encountered an issue while processing your request:\n`{}`\n\n"
+					"Please review your input and try again.\n\n"
+					"*Time:* `{}` UTC\n\n"
+					"Best regards,\nYour Team :robot_face:"
+				).format(specific_error, timestamp)
+				return error_message, None
+
+			elif service_type == 'msteams':
+				error_message = (
+					"Warning: Hello!\n\n"
+					"We encountered following issue while processing your request.\n\n`{}`\n\n"
+					"Please review your input and try again.\n\n"
+					"Time: `{}` UTC\n\n"
+					"Best regards,\nYour Team"
+				).format(specific_error, timestamp)
+				return error_message, None
+
+		except Exception as e:
+			# Log any unexpected exceptions that might occur while generating error message
+			L.warning("An unexpected error occurred while generating error message: {}".format(str(e)))
```

Comment:
This is `L.exception` and don't do `.format(e)` - that's suitable only for certain exception types.